File: Algorithms/2. Between Two Sets/solution.lua


--
-- Complete the 'getTotalX' function below.
--
-- The function is expected to return an INTEGER.
-- The function accepts following parameters:
--  1. INTEGER_ARRAY a
--  2. INTEGER_ARRAY b
--

function getTotalX(a, b)
    local count = 0
    
    local all_factors = false
    local is_all_factor = false
    
    table.sort(b)
    
    max_num = b[#b]
    
    for i = 1, max_num, 1 do
        for j = 1, #a, 1 do
            if i % a[j] ~= 0 then
                all_factors = false
                break
            end     
            all_factors = true
        end
            
        for k = 1, #b, 1 do
            if b[k] % i ~= 0 then
                is_all_factor = false
                break
            end
            is_all_factor = true
        end
        
        if all_factors and is_all_factor then
            count = count + 1
        end
    end
    
    return count
end

local fptr = io.open(os.getenv("OUTPUT_PATH"), "w")

local firstmultipleinput = {}
for token in string.gmatch(io.stdin:read("*l"):gsub("%s+$", ""), "[^%s]+") do
    table.insert(firstmultipleinput, token)
end

local n = tonumber(firstmultipleinput[1]:match("^%s*(.-)%s*$"))

local m = tonumber(firstmultipleinput[2]:match("^%s*(.-)%s*$"))

local arr = {}

for token in string.gmatch(io.stdin:read("*l"):gsub("%s+$", ""), "[^%s]+") do
    table.insert(arr, tonumber(token))
end

local brr = {}

for token in string.gmatch(io.stdin:read("*l"):gsub("%s+$", ""), "[^%s]+") do
    table.insert(brr, tonumber(token))
end

local total = getTotalX(arr, brr)

fptr:write(total, "\n")

fptr:close()

File: Algorithms/2. Between Two Sets/solution.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'getTotalX' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY a
#  2. INTEGER_ARRAY b
#

def getTotalX(a, b):
    count = 0
    
    all_factors = False
    is_all_factor = False
    
    max_num = max(b)
    
    for i in range(1, max_num+1):
        for j in range(len(a)):
            if i % a[j] != 0:
                all_factors = False
                break
            all_factors = True
        
        for k in range(len(b)):
            if b[k] % i != 0:
                is_all_factor = False
                break
            is_all_factor = True
        
        if all_factors and is_all_factor:
            count += 1
            
    return count   

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])

    m = int(first_multiple_input[1])

    arr = list(map(int, input().rstrip().split()))

    brr = list(map(int, input().rstrip().split()))

    total = getTotalX(arr, brr)

    fptr.write(str(total) + '\n')

    fptr.close()
